packages/console/src/components/Entities/EntityDetails.tsx

```diff
@@ -57,58 +63,70 @@ interface EntityDetailsProps {
   const sections = entitySections[id.resourceType];
-  const project = useProject(id.project);
+  const [project] = useProject(id.project);
   const styles = useStyles();
   const { chartIds, onToggle, clearCharts } = useChartState();
 
   return (
-    <WaitForData {...project}>
-      <EntityDetailsHeader
-        project={project.value}
-        id={id}
-        launchable={!!sections.launch}
-      />
+    <Grid container direction="column" className={styles.entityDetailsWrapper}>
+      {project?.id ? (
+        <>
+          <EntityDetailsHeader
+            project={project}
+            id={id}
+            launchable={!!sections.launch}
+          />
 
-      <div className={styles.metadataContainer}>
-        {sections.description ? (
-          <div className={styles.descriptionContainer}>
-            <EntityDescription id={id} />
-          </div>
-        ) : null}
-        {!sections.inputs && sections.schedules ? (
-          <div className={styles.schedulesContainer}>
-            <EntitySchedules id={id} />
+          <div className={styles.metadataContainer}>
+            {sections.description ? (
+              <div className={styles.descriptionContainer}>
+                <EntityDescription id={id} />
+              </div>
+            ) : null}
+            {!sections.inputs && sections.schedules ? (
+              <div className={styles.schedulesContainer}>
+                <EntitySchedules id={id} />
+              </div>
+            ) : null}
           </div>
-        ) : null}
-      </div>
 
-      {sections.inputs ? (
-        <div className={styles.inputsContainer}>
-          <EntityInputs id={id} />
-        </div>
-      ) : null}
+          {sections.inputs ? (
+            <div className={styles.inputsContainer}>
+              <EntityInputs id={id} />
+            </div>
+          ) : null}
 
-      {sections.versions ? (
-        <div className={styles.versionsContainer}>
-          <EntityVersions id={id} />
-        </div>
-      ) : null}
+          {sections.versions ? (
+            <div className={styles.versionsContainer}>
+              <EntityVersions id={id} />
+            </div>
+          ) : null}
 
-      <EntityExecutionsBarChart
-        onToggle={onToggle}
-        chartIds={chartIds}
-        id={id}
-      />
-
-      {sections.executions ? (
-        <div className={styles.executionsContainer}>
-          <EntityExecutions
+          <EntityExecutionsBarChart
+            onToggle={onToggle}
             chartIds={chartIds}
             id={id}
-            clearCharts={clearCharts}
           />
-        </div>
-      ) : null}
-    </WaitForData>
+
+          {sections.executions ? (
+            <div className={styles.executionsContainer}>
+              <EntityExecutions
+                chartIds={chartIds}
+                id={id}
+                clearCharts={clearCharts}
+              />
+            </div>
+          ) : (
```

Comment:
I would avoid use of multiple ? :

If possible, we should define either a new component to handle specific states or define a function `renderExecutions` that does this logic using an if else to make it easy to read and test for